platform-services.md

```diff
@@ -0,0 +1,215 @@
+transferring handles to other objects along with messages. You are responsible for encoding and
+decoding messages sent through this mechanism.
+
+## Get the `hello_services` sample {#the-hello_services-sample}
+
+The Flutter SDK includes a reference sample project called
+[`hello_services`](https://github.com/flutter/flutter/tree/master/examples/hello_services)
+that demonstrates how to connect to the geolocation service within an iOS or Android application
+by using the `FlutterView` interface.
+
+The `hello_services` project comes with helper code to handle the passing of
+JSON-formatted messages to invoke platform services. If your app needs access to functionality
+provided by platform-specific iOS or Android services, we strongly encourage you to
+[download](https://github.com/flutter/flutter/tree/master/examples/hello_services) and use this
+sample project as the starter code to build your own Flutter app.
+
+The following diagram shows how the `hello_services` app uses the `FlutterView` interface to
+send and receive messages to platform and third-party services on iOS and Android.
+
+ ![hello_services architecture](/images/hello_services.png)
+
+**Fig 1.** Architecture of the `hello_services` sample.
+
+The Dart, iOS, and Android-specific components are related as follows:
+
+* In the Dart code, the sample app imports the
+[`flutter/services.dart`](http://docs.flutter.io/flutter/services/services-library.html)
+package. This package defines the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class,
+which provides a bridge to send and listen for JSON-encoded messages exchanged with a platform
+service on a <em>host</em> app (that is, the iOS or Android-specific parts of your app). The
+<code>hello_services</code> app uses the static methods of the
+[`HostMessages`](http://docs.flutter.io/flutter/services/HostMessages-class.html) class
+in the [`State`](http://docs.flutter.io/flutter/material/State-class.html) object of a
+stateful widget to make services requests, then updates the UI accordingly.
+
+* On iOS, the sample app imports the Flutter framework library (`Flutter.h`) as a
+dependency. The library defines the <code>FlutterViewController</code> class, which is a
+[`UIViewController`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIViewController_Class/)
+that provides a bridge from the
+[`UIKit`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIKit_Framework/)
+view hierarchy to the Flutter app. The `hello_services` app sets an instance of
+[`FlutterViewController`](https://github.com/flutter/engine/blob/master/sky/shell/platform/ios/framework/Headers/FlutterViewController.h)
+as the root view controller of its [`UIWindow`](https://developer.apple.com/library/ios/documentation/UIKit/Reference/UIWindow_Class/).
+
+* On Android, the sample imports the `flutter.jar` file as a dependency. This JAR
```

Comment:
This bullet is not parallel with the previous bullet.  `flutter.jar` is the build artifact, not the interface, but `Flutter.h` is the interface, not the build artifact (which is `Flutter.framework`).